app/uk/gov/hmrc/cdsreimbursementclaimfrontend/services/FeatureSwitchService.scala

```diff
@@ -22,61 +22,43 @@ import play.api.Configuration
 import play.api.mvc.Results.NotFound
 import play.api.mvc._
 import uk.gov.hmrc.cdsreimbursementclaimfrontend.config.ErrorHandler
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.models.Feature
 
 import scala.concurrent.{ExecutionContext, Future}
+import java.util.concurrent.ConcurrentHashMap
 
 @Singleton
-class FeatureSwitchService @Inject() (configuration: Configuration) {
+class FeatureSwitchService @Inject() (
+  val configuration: Configuration,
+  errorHandler: ErrorHandler,
+  cc: MessagesControllerComponents
+) {
 
-  def of(name: String): Option[FeatureName] = Seq(
-    BulkMultiple,
-    BulkClaim,
-    NorthernIreland,
-    RejectedGoods
-  ).find(_.name === name)
+  private val features: ConcurrentHashMap[String, Boolean] =
+    new ConcurrentHashMap[String, Boolean]()
 
-  sealed trait FeatureName extends Product with Serializable {
+  def enable(feature: Feature): Boolean =
+    features.put(feature.name, true)
 
-    val name: String
-    val confPropertyName: String   = s"features.$name"
-    val systemPropertyName: String = s"features.$name"
+  def disable(feature: Feature): Boolean =
+    features.put(feature.name, false)
 
-    def isEnabled(): Boolean =
-      sys.props.get(systemPropertyName).map(_.toBoolean).getOrElse {
-        configuration.getOptional[Boolean](confPropertyName).getOrElse(false)
-      }
+  def isEnabled(feature: Feature): Boolean =
+    Option(features.get(feature.name))
+      .orElse(sys.props.get(s"features.${feature.name}").map(_.toBoolean))
+      .orElse(configuration.getOptional[Boolean](s"features.${feature.name}"))
+      .getOrElse(false)
 
-    def enable(): Unit =
-      setProp(true)
+  def hideIfNotEnabled(feature: Feature): ActionBuilder[Request, AnyContent] with ActionFilter[Request] =
+    new ActionBuilder[Request, AnyContent] with ActionFilter[Request] {
 
-    def disable(): Unit =
-      setProp(false)
+      def filter[A](input: Request[A]): Future[Option[Result]] = Future.successful {
+        Option(isEnabled(feature)).filter(_ === false) map (_ => NotFound(errorHandler.notFoundTemplate(input)))
```

Comment:
Isn't the possibility of failure handled in the isEnabled method?

Reply:
I'm reusing here an existing impl, but yes, we can improve it

Reply:
@steve-edmans addressed your comments